### PR TITLE
- special rollback for transactional server (bsc#1172273)

### PR DESCRIPTION
**Pull request by aschnell**
This PR specializes the rollback command for transactional servers. The detection of transactional servers is done automatic but can be overruled by a global command line option.

The error handling of command line options is also change to use exception reducing the LOC significantly.

See https://trello.com/c/7wOPtMOn/1892-3-ostumbleweed-p5-1172273-snapper-rollback-considered-harmful-in-a-transactional-installation and https://bugzilla.suse.com/show_bug.cgi?id=1172273.

### CODE REVIEW

**Comment by aschnell** [comment]
We will deal later with the Travis failures.

File: Dockerfile.tumbleweed [hunk 1]
```diff
@@ -9,16 +9,15 @@ FROM opensuse/${BUILD_FLAVOR}
...
 # see https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#run
 # https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/build-cache
 # why we need "zypper clean -a" at the end
-RUN zypper --non-interactive in --no-recommends --force-resolution ruby && zypper clean -a
+RUN zypper --non-interactive in --no-recommends --force-resolution util-linux && zypper clean -a
 
 # add the YaST repository - we need rubygem-coveralls-lcov
 RUN zypper ar -f http://download.opensuse.org/repositories/YaST:/Head/openSUSE_Tumbleweed/ yast_tw
-RUN zypper ar -f http://download.opensuse.org/repositories/YaST:/Head/openSUSE_Leap_15.1_Update/ yast_lp
+RUN zypper ar -f http://download.opensuse.org/repositories/YaST:/Head/openSUSE_Leap_15.2_Update/ yast_lp
```
Comment (lslezak) [inline Dockerfile.tumbleweed:20]: Um, we have "openSUSE_Leap_15.2" target only (which already points to openSUSE:Leap:15.2:Update), see https://build.opensuse.org/repositories/YaST:Head

File: client/GlobalOptions.h [hunk 2]
```diff
@@ -38,132 +40,64 @@ namespace snapper
...
+	    bool iso() const { return _iso; }
+	    bool no_dbus() const { return _no_dbus; }
+	    bool version() const { return _version; }
+	    bool help() const { return _help; }
+	    TableLineStyle table_style() const { return _table_style; }
+	    OutputFormat output_format() const { return _output_format; }
+	    string separator() const { return _separator; }
+	    string config() const { return _config; }
+	    string root() const { return _root; }
+	    Ambit ambit() { return _ambit; }
```
Comment (shundhammer) [inline client/GlobalOptions.h:62]: Not `Ambit ambit() const` ?

**Comment by shundhammer** [comment]
Just the one thing with the `const' getter, otherwise LGTM.

File: client/GlobalOptions.h (hunk 2 above)
Comment (aschnell) [reply client/GlobalOptions.h]: Sure.

**Pull request merged**